content/en/blog/_posts/2025-01-22-kube-scheduler-simulator/index.md

```diff
@@ -0,0 +1,138 @@
+
+It is useful for casual users who employ scheduling constraints (e.g., inter-Pod affinity)
+and experts who extend the scheduler with custom plugins.
+
+## Motivation
+
+The scheduler often appears as a black box, 
+composed of many plugins that each contribute to the scheduling decision-making process from their unique perspectives. 
+Understanding its behavior can be challenging due to the multitude of factors it considers. 
+Even if a Pod seems to be scheduled as expected in your simple test cluster, 
+it may be for a different reason than what you imagine, 
+which could result in unexpected scheduling results in a large production environment.
+
+Also, testing your scheduler is a complex challenge.
+There are countless patterns of operations executed within a real cluster, making it impractical to anticipate every scenario with a finite number of tests. 
+More often than not, bugs are discovered only when the scheduler is deployed in an actual cluster.
+Actually, many bugs are reported by users after we ship the release, and the bugs are found
+even in the upstream kube-scheduler. 
+
+Having a development or sandbox environment for testing the scheduler — or, indeed, any Kubernetes controllers — is a common practice.
+However, this approach falls short of capturing all the potential scenarios that might arise in a production cluster 
+because a development cluster is often much smaller with notable differences in workload sizes and scaling dynamics.
+It never sees the exact same use or exhibits the same behavior as its production counterpart.
+
+The scheduler simulator aims to solve those problems.
+It enables users to test their scheduling constraints, scheduler configurations, 
+and custom plugins while checking every detailed part of scheduling decisions.
+It also allows you to create a simulated cluster environment, where you can test your scheduler
+with the same resources as your production cluster without affecting your actual workloads.
+
+## Features of the Kube-scheduler-simulator
+
+The simulator’s core feature is its ability to expose the scheduler's internal decisions.
+The scheduler operates based on the [scheduling framework](/docs/concepts/scheduling-eviction/scheduling-framework/), 
+utilizing various plugins at different extension points; 
+filter nodes (Filter phase), score nodes (Score phase), and ultimately determine the best node for the Pod.
+
+The simulator allows users to create Kubernetes resources and observe how each plugin influences the scheduling decisions for Pods.
+This visibility helps users understand the scheduler’s workings and define appropriate scheduling constraints.
+
+{{< figure src="/images/blog/2025-01-22-kube-scheduler-simulator/simulator.png" alt="Screenshot of the simulator web frontend" title="The simulator web frontend" >}}
+
+Inside the simulator, a debuggable scheduler runs instead of the vanilla scheduler. 
```

Comment:
what is vanilla scheduler in this context? Does it mean that the decision will be different between those two? If you mean the same scheduler, but in debug mode - it may be a better way to describe what is hapenning. Otherwise it is very confusing to read. Why I would want some other scheduler decisions for my investigation?